kernel/configs/fragments/4.19.x/common/base.conf

```diff
@@ -0,0 +1,36 @@
+# Basic necessary items!
+
+CONFIG_SMP=y
+CONFIG_HYPERVISOR_GUEST=y
+CONFIG_PARAVIRT=y
+CONFIG_KVM_GUEST=y
+# Note, no nested VM support enabled here
+
+# Note, no virt enabled baloon yet
+CONFIG_INPUT=y
+CONFIG_CPU_FREQ=y
+CONFIG_PRINTK=y
+# We use this for metrics!
+CONFIG_PRINTK_TIME=y
+CONFIG_UNIX98_PTYS=y
+CONFIG_FUTEX=y
+CONFIG_HIGH_RES_TIMERS=y
+CONFIG_GENERIC_MSI_IRQ_DOMAIN=y
+CONFIG_GENERIC_MSI_IRQ=y
+CONFIG_LEGACY_VSYSCALL_NONE=y
+CONFIG_NO_HZ=y
+CONFIG_NO_HZ_FULL=y
+CONFIG_POSIX_MQUEUE=y
+CONFIG_POSIX_TIMERS=y
+CONFIG_PROC_SYSCTL=y
+
+CONFIG_SHMEM=y
+
+# For security...
+CONFIG_RELOCATABLE=y
```

Comment:
@grahamwhaley - Does the tooling support "inline comments"? If so, we could consider moving all comments like this to the right:

```
CONFIG_RELOCATABLE=y # For security...
```

That would make the fragments "more structured" allowing easier parsing of per config option details. If this is supported, we should require that such comments *only* occur on the same line as the config option itself. That does imply we may end up with some long lines, but... tough ;)

Reply:
In fact, how about:

```
CONFIG_RELOCATABLE=y # tags=security,foo,bar:we need this to avoid killing unicorns.
```

Reply:
afaik, the [core script](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/scripts/kconfig/merge_config.sh) from the kernel we use to do the fragment merge uses the standard Kconfig file format - which afaik does not support tail-end comments (well, I've never seen one in a kernel config file ;-) ).
And as for YAML etc. - I know you'd love this to be all self checking etc., but given that the whole process (from the user running say a `menuconfig` on the kernel and looking at their .config and then diffing/extracting `CONFIG_*=y` lines and creating a new fragment file) is all in Kconfig format - I would say we really don't want to invent or embellish our own format for this - this is all about kernel config files - we should stick with that format.

The tool itself does report values that are overridden and values that are set but then do not appear in the final .config file. Sadly right now we throw that info away (`> /dev/null`), as the shell function it is called in returns the config path with an `echo`. yeah, we should probably capture that output and grep it for the two error lines, and balk out if we find any - I'll look at adding that.

Reply:
maybe this can be a longer play @jodh-intel -- I'd rather stick to upstream tooling at least as a first step?

Reply:
Well, that does mean we'll just be storing up more of a problem for ourselves in we ever do change format as we'll then have to pick over all the config options we'll have built up by then and try to work out from the PRs what the intent was. Whereas if we do it now, we can mandate that "groups" of config options atleast are annotated somehow to justify whey they are set / unset. Going through that exercise was valuable for qemu (see https://github.com/kata-containers/packaging/blob/master/scripts/configure-hypervisor.sh#L193).

However, for the sake of getting this PR landed soon, I understand that my idea to create a bunch of extra work now rather than later is not going to be that popular :smile: